### PR TITLE
Mongo_db Added transfer_operation implementation #703

### PR DESCRIPTION
**Pull request by maslenitsa93**

### CODE REVIEW

File: plugins/mongo_db/mongo_db_state.cpp [hunk 1]
```diff
@@ -239,7 +242,40 @@ namespace mongo_db {
     }
 
     auto state_writer::operator()(const transfer_operation& op) -> result_type {
-        
+        auto oid = std::string(op.from).append("/").append(op.to)
+            .append("/").append(state_block.timestamp);
```
Comment (afalaleev) [inline plugins/mongo_db/mongo_db_state.cpp]: This isn't unique.
There are no limits to transfer more than 1 time per block from one account to another.
Transfers don't have update operations, so _id can be calculated by Mongo.

File: plugins/mongo_db/mongo_db_state.cpp [hunk 2]
```diff
@@ -239,7 +242,40 @@ namespace mongo_db {
...
+        auto oid_hash = fc::sha1::hash(oid).str().substr(0, 24);
```
Comment (afalaleev) [inline plugins/mongo_db/mongo_db_state.cpp]: Why do you do it manually? Is it possible to use one function for all cases like `format_oid`?

Comment (maslenitsa93) [reply plugins/mongo_db/mongo_db_state.cpp]: For which case? For "fc::sha1::hash..." - yes, but for "oid = ..." - no, it's different for different object types.

Comment (afalaleev) [reply plugins/mongo_db/mongo_db_state.cpp]: The `format_oid` contains expression for calculating oid from string. 
But in this place it seems like same magic.
I request to create a function which calculates hash - and this function should be used in places where it is needed - in format_oid, and in all places where this hash is used for key in bmi.

**Pull request closed without merging**